### PR TITLE
Bug 1694107 - Time range dropdown not synced with Graphs view

### PR DESCRIPTION
**Pull request by beatrice-acasandrei**
Steps to reproduce can be found here: https://bugzilla.mozilla.org/show_bug.cgi?id=1694107
Also, when reviewing please take a look at @ionutgoldan's comment.

### CODE REVIEW

File: ui/perfherder/graphs/TestDataModal.jsx [hunk 1]
```diff
@@ -95,6 +95,11 @@ export default class TestDataModal extends React.Component {
       this.processOptions(true);
     }
 
+    if (timeRange !== prevProps.timeRange) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({ innerTimeRange: this.props.timeRange });
```
Comment (ionutgoldan) [inline ui/perfherder/graphs/TestDataModal.jsx:100]: We could simply use `timeRange`.

**Review by ionutgoldan — CHANGES_REQUESTED** [review]
There's still a bug in our logic.

#### Steps to reproduce

1. Go to default page of Graphs view.
2. Click on **Add test data**.
3. Select **Last 60 days**.
4. Click the **X** button to close the modal.
5. Click again on **Add test data**.

Expected:
Time range from modal resets to the default value.

What we got instead:
Time range keeps its new value.

File: ui/perfherder/graphs/TestDataModal.jsx [hunk 2]
```diff
@@ -370,6 +370,7 @@ export default class TestDataModal extends React.Component {
       },
       this.props.toggle,
     );
+    this.setState({ innerTimeRange: this.props.timeRange });
```
Comment (ionutgoldan) [inline ui/perfherder/graphs/TestDataModal.jsx:373]: We already have a `setState()` call. We could simply make use of it.

File: ui/perfherder/graphs/TestDataModal.jsx [hunk 3]
```diff
@@ -365,6 +373,7 @@ export default class TestDataModal extends React.Component {
       },
```
Comment (ionutgoldan) [inline ui/perfherder/graphs/TestDataModal.jsx:373]: I was actually referring to adding an entry here, in this existing `setState()`.

**Review by ionutgoldan — CHANGES_REQUESTED** [review]

**Review by airimovici — APPROVED** [review]

**Review by ionutgoldan — APPROVED** [review]

**Pull request merged**